tests/back/api/test_global_explanation_api.py

```diff
@@ -0,0 +1,204 @@
+    script_dir = os.path.dirname(__file__)
+    test_dataset = "iris.csv"
+    abs_file_path = os.path.join(script_dir, test_dataset)
+    with open(abs_file_path, "rb") as csv_file:
+        response = client.post(
+            "/api/v1/dataset/",
+            data={
+                "params": """{  "task_name": "TabularClassificationTask",
+                                    "dataloader": "CSVDataLoader",
+                                    "dataset_name": "test_iris",
+                                    "outputs_columns": ["Species"],
+                                    "splits_in_folders": false,
+                                    "splits": {
+                                        "train_size": 0.5,
+                                        "test_size": 0.2,
+                                        "val_size": 0.3,
+                                        "seed": 42,
+                                        "shuffle": false,
+                                        "stratify": false
+                                    },
+                                    "dataloader_params": {
+                                        "data_key": "data"
+                                    }
+                                }""",
+                "url": "",
+            },
+            files={"file": ("filename", csv_file, "text/csv")},
+        )
+    assert response.status_code == 201, response.text
+    dataset = response.json()
+    yield dataset["id"]
+    response = client.delete(f"/api/v1/dataset/{dataset['id']}")
+    assert response.status_code == 204, response.text
+
+
+@pytest.fixture(scope="module", name="experiment_id")
+def fixture_experiment_id(session: sessionmaker, dataset_id: int):
+    db = session()
+    dummy_experiment = Experiment(
+        dataset_id=dataset_id,
+        task_name="TabularClassificationTask",
+        name="Test Experiment",
+    )
+    db.add(dummy_experiment)
+    db.commit()
+    db.refresh(dummy_experiment)
+    yield dummy_experiment.id
+
+    db.delete(dummy_experiment)
+    db.commit()
+
+
+@pytest.fixture(scope="module", name="run_id_1")
+def fixture_run_id_1(session: sessionmaker, experiment_id: int):
+    db = session()
+    run = Run(
+        experiment_id=experiment_id,
+        model_name="RandomForestClassifier",
+        parameters={},
+        name="Run",
+    )
+    db.add(run)
+    db.commit()
+    db.refresh(run)
+
+    yield run.id
+
+    db.delete(run)
+    db.commit()
+    db.close()
+
+
+@pytest.fixture(scope="module", name="run_id_2")
+def fixture_run_id_2(session: sessionmaker, experiment_id: int):
+    db = session()
+    run = Run(
+        experiment_id=experiment_id,
+        model_name="SVC",
+        parameters={},
+        name="Run",
+    )
+    db.add(run)
+    db.commit()
+    db.refresh(run)
+
+    yield run.id
+
+    db.delete(run)
+    db.commit()
+    db.close()
+
+
+def test_create_global_explanation(client: TestClient, run_id_1: int, run_id_2):
```

Comment:
Here where do you pass the categorical variables (that are declared in the partial dependency explain signature)?